### DISCUSSION TITLE
Idea for improvement.

### DISCUSSION THREAD

**Discussion by onesine** [sutandojs/sutando#63] [Ideas]
Hello 👋 👋 👋 @kiddyuchina  ,

First of all, I would like to thank you for the incredible work you've done on **Sutando**. I enjoyed working with the **Laravel framework** for a long time, and now that I'm exploring **JavaScript**, I'm still looking for the same comfort I had with **Laravel**. That's how I started looking for an **ORM** similar to **Eloquent**, and that led me to discover **Sutando**. I believe this project should be more widely known and given greater visibility.

There are a few points I'd like to discuss with you. I'll list them here first:

- How can we verify that the `database` connection is established? I'd like my web server to start only if the connection to the `database` is successful.

- In the [rest-express](https://github.com/sutandojs/sutando-examples/tree/main/typescript/rest-express) example, for `migrations` and `seeders`, we need to manually add the `database` connection. Wouldn't it be possible to have a configuration file that returns a `database` connection instance? Then, in `migrations` or `seeders`, we could call this file to retrieve the instance and check the connection before proceeding with creating the schema or executing the `seeders`.

- In a project using **TypeScript** with `type: "module"` in the `package.json`, when running the `sutando init` command, a `sutando.config.js` file is generated with `module.exports` inside. However, as far as I know, `module.exports` is not supported with the module type, and the file is not a `.ts` file either.

I'm open to discussion and look forward to hearing from you.

**Comment by kiddyuchina** [comment]
Hi, Thank you so much for your support and recognition of Sutando! I'm delighted to hear that you found Sutando while exploring JavaScript and that it provides you with a similar comfort to Laravel. Your feedback is very important to us, and here are my responses to the points you mentioned:

- You can verify the connection by executing a simple query. For example:
```js
 try {
   const connection = sutando.connection();
   await connection.raw('SELECT 1+1 AS result');
   console.log('Connection has been established successfully.');
 } catch (error) {
   console.error('Unable to connect to the database:', error);
 }
```
  This ensures that your web server will only start if the connection to the database is successful. Subsequent versions can add a method to connection to help users perform this query.

- Your suggestion is excellent. I admit that I was a little lazy when writing examples. We should include a configuration file in the example so that the database connection instance can be directly referenced in migration or seed.

- If you have set `"type": "module"` in your project's `package.json`, you can change the extension of sutando.config.js and migration files to `.cjs`. This will ensure that migration commands execute correctly. In the future, sutando CLI can add file suffix options when generating config and migration files to suit different projects.

Thank you very much for your valuable suggestions! If you have more ideas or questions, feel free to discuss them with me anytime.
